### PR TITLE
same counts setup

### PR DESCRIPTION
**Pull request by msultan**
Based upon #718, I refactored the build counts setup into a new mixin. Naming could be better but I wasn't sure what else to call it. Also added a test that compares the counts across 3 models and gets the same answer. Might need to construct a test that throws some states away but this should suffice for now i think.

### CODE REVIEW

**Comment by mpharrigan** [comment]
Very nice!

File: msmbuilder/msm/ratematrix.py [hunk 1]
```diff
@@ -131,30 +132,6 @@ def __init__(self, lag_time=1, n_timescales=None, ergodic_cutoff=1,
         self.right_eigenvectors_ = None
         self.percent_retained_ = None
```
Comment (mpharrigan) [inline msmbuilder/msm/ratematrix.py]: don't you need to add in a call to _build_counts in continuoustimemsm

Comment (mpharrigan) [inline msmbuilder/msm/ratematrix.py]: nevermind, i saw it

File: msmbuilder/msm/core.py [hunk 2]
```diff
@@ -205,6 +181,32 @@ def inverse_transform(self, sequences):
...
+                _strongly_connected_subgraph(raw_counts, ergodic_cutoff,
+                                             self.verbose)
+            self.mapping_ = _dict_compose(mapping, mapping2)
+        else:
+            # no ergodic trimming.
+            self.countsmat_ = raw_counts
+            self.mapping_ = mapping
+            self.percent_retained_ = 100
+        self.n_states_ = self.countsmat_.shape[0]
+        return ergodic_cutoff
```
Comment (mpharrigan) [inline msmbuilder/msm/core.py]: why does this return `ergodic_cutoff`? Maybe just don't return anything

File: msmbuilder/msm/msm.py [hunk 3]
```diff
@@ -154,7 +156,7 @@ def fit(self, sequences, y=None):
         None will not be counted. The mapping_ attribute will not include the
         NaN or None.
         """
-        _ = self._setup(sequences)
+        _ = self._build_counts(sequences)
```
Comment (mpharrigan) [inline msmbuilder/msm/msm.py]: might as well completely omit the `_ = `

(the reason why this is sometimes used is if you are unpacking a tuple and don't care about one of the values)
```python
five, _ = (5, SomeCrazyObject())
print(five)
```

File: msmbuilder/msm/core.py (hunk 2 above)
Comment (msultan) [inline msmbuilder/msm/core.py]: Thats true, let me see if the tests all pass and i can push another change in.

File: msmbuilder/msm/msm.py (hunk 3 above)
Comment (msultan) [inline msmbuilder/msm/msm.py]: I think this would change once i update the ```_build_counts ``` method.

**Comment by mpharrigan** [comment]
And you can add an appropriate entry to `changelog.rst` explaining how counts used to be inconsistent and now they will always be consistent

**Comment by msultan** [comment]
also yay/nay on the naming scheme?

**Comment by mpharrigan** [comment]
Seems fine to me (aka yay)

**Comment by msultan** [comment]
maybe we should just call it MSMMixin and move the sample_discrete method there too?

**Comment by msultan** [comment]
why the hell is ```test_kmedoids.py``` failing on py3.5?

**Comment by mpharrigan** [comment]
> maybe we should just call it MSMMixin and move the sample_discrete method there too?
-1 for now

**Comment by msultan** [comment]
@mpharrigan any idea as to why this is failing on py3.4 but passing on py2.7 and py3.5?

**Comment by mpharrigan** [comment]
If you create a py34 conda env can you reproduce locally?

I'll restart the tests but it may be a real issue

**Comment by mpharrigan** [comment]
eh probably just something weird

**Pull request merged**